FWCore/Framework/interface/ModuleTypeResolverMaker.h

```diff
@@ -0,0 +1,38 @@
+#ifndef FWCore_Framework_ModuleTypeResolverMaker_h
+#define FWCore_Framework_ModuleTypeResolverMaker_h
+
+#include "FWCore/Framework/interface/ModuleTypeResolverBase.h"
+
+namespace edm {
+  class ParameterSet;
+
+  /**
+   * This class hierarchy implements an abstract factory pattern for
+   * the creation of module type resolver objects (derived from
+   * ModuleTypeResolverBase). This pattern allows the use of
+   * information in the module PSet in the module type resolver in a
+   * way that the PSet is parsed only once per module instance.
```

Comment:
Maybe an example of why one would want to use info in the module PSet?

Reply:
How about now?